### DISCUSSION TITLE
Never struggle with finding beautiful themes for shadcn again

### DISCUSSION THREAD

**Discussion by OrelsProjects** [shadcn-ui/ui#6229] [Show and tell]
Hey all 👋

As I was struggling with finding beautiful, hand picked themes for my shadcn styling.
I decided to take matters into my own hands and build a tool that has dozens of beautiful palettes, the shadcn way.

https://www.shadcn.studio/

Enjoy, completely free!

Let me know what you think :)

**Comment by bookwormapp2** [comment]
Wow, well done.
Exactly what I needed.

The 100% generator is extreme 😅

**Reply by OrelsProjects** [reply]
Then I did a good job :)

**Comment by MithrilMan** [comment]
Nice idea!
I'm new to shadcn but from a brief look it seems your theme lacks support for some components (e.g. the sidebar ins't styled)

**Reply by OrelsProjects** [reply]
There's no lack of support :)

I just show you some generic components to give you a feeling of what the palette is like.
If you copy and paste the palette into your global.css, it will apply to your sidebar as well :)

**Reply by OrelsProjects** [reply]
Let me know if you have any more questions or need any help implementing it.

**Reply by MithrilMan** [reply]
May be beacause I'm into the library just since few days but I'm pretty sure that when I added the sidebar component in my solution (I use the CLI to add components I need one by one) it added this styles to my index.scss:

```
--sidebar-background: 0 0% 98%;
--sidebar-foreground: 240 5.3% 26.1%;
--sidebar-primary: 240 5.9% 10%;
--sidebar-primary-foreground: 0 0% 98%;
--sidebar-accent: 240 4.8% 95.9%;
--sidebar-accent-foreground: 240 5.9% 10%;
--sidebar-border: 220 13% 91%;
--sidebar-ring: 217.2 91.2% 59.8%;
```

Btw this is missing on the official theme builder in the official shadcn-ui website

**Reply by MithrilMan** [reply]
Ok I see the documentation explicitly talks about sidebar theming here:
https://ui.shadcn.com/docs/components/sidebar#theming

While I agree sidebar shouldn't probaby use the same primary/secondary color, I think should be included in the theming tool (talking about the official one, and yours of course if you please so)

**Reply by OrelsProjects** [reply]
Got it!

I added it :) 
go ahead and check it out, let me know what you think!

https://www.shadcn.studio

**Reply by MithrilMan** [reply]
nice, since I know how much feedback is importants and how much makes happy the developer, here some others:

Improve selection clues (Usability)
It's not clear which theme I've selected, either make it clear in the dialog (e.g. highlight selected theme) or write it somewhere (the app header is almost empty so could be a candidate)

Some themed sidebar have contrast problems, e.g. select the DaisyUI Lemonade theme and look at the sidebar (hint, the icon on the top isn't visible because of lack of contrast)
e.g. dark theme 
![image](https://github.com/user-attachments/assets/a35ac62c-4c3c-4627-83cd-ed38f113a575)

light theme
![image](https://github.com/user-attachments/assets/41b62355-7e72-436d-8d38-bcfb219e4caa)

that problem is visible on different themes and sometimes only in one variant (e.g. this dark ByteDispute)
![image](https://github.com/user-attachments/assets/e3b6fba3-7e7f-4c5b-9898-674abcb34ad6)


good job

**Reply by OrelsProjects** [reply]
@MithrilMan I appreciate the feedback. I love it.

The selected theme visibility is important, I agree. I added it to the header.

![image](https://github.com/user-attachments/assets/70a4fe12-3268-439c-979e-831c4284796c)
![image](https://github.com/user-attachments/assets/8564f2cc-8380-4db8-b7a4-9feaa3eec58a)
![image](https://github.com/user-attachments/assets/48889fb6-cf0c-419c-8ab2-0fab3ac1d319)
![image](https://github.com/user-attachments/assets/de6fc4f5-ef4a-4d58-8481-3f7c748d2d71)

(It's fixed for all themes, shared the specific ones you shared as problematic :))

Let me know what you think.

And again, thank you man 👍

**Comment by OrelsProjects** [comment]
Improved the UI :)
Enjoy!

**Comment by OrelsProjects** [comment]
Some of the themes had problems with light mode.
Fixed :)

**Reply by vhfferreira** [reply]
Thanks. Great work!

**Comment by andrewmkon** [comment]
What's interesting is that it's impossible to find Material Design theme for shadcn on the internet, and your theme collection lacks it as well.

**Reply by OrelsProjects** [reply]
Great point! I will add it :)

**Comment by kamami** [comment]
![image](https://github.com/user-attachments/assets/7e607208-39a4-4e2f-aa2d-c966c015bcf2)
Just FYI
